### PR TITLE
#167840707 signup with postgres db

### PR DESCRIPTION
**Pull request by WinnersProx**
#### What does this PR do?
- implement signing up with Postgres DB
#### Description of Task to be completed?
- version of the application
#### How should this be manually tested?
- this will be tested making a post request to [here](https://way-farer-app-rest.herokuapp.com/api/v2)
#### What are the relevant pivotal tracker stories?
- https://www.pivotaltracker.com/story/show/167417680

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,26 +1,24 @@
...
+  error  : `The requested resource was not found on the server, Kindly read documentation here : ${docsUrl}`
+}));
 
 app.listen(PORT, () => {
-    console.log(`WayFarer server has been started on port:${PORT}`)
-})
+  console.log(`WayFarer server has been started on port:${PORT}`)
+});
 
 export default app
```
Comment (oesukam) [inline index.js]: Empty line required

File: server/v1/config/db.config.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+const pool = new Pool({ 
+	user: 'me',
+  	host: 'localhost',
+  	database: 'way_farer_db',
+  	password: 'winners1',
+  	port: 5432
+});
+
+export default pool;
```
Comment (oesukam) [inline server/v1/config/db.config.js]: Empty line required

File: server/v1/models/db/schema.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+${bookingsTable}
+`;
+
+(async () => {
+    try {
+        await pool.query(queryString);
+    } catch (error) {
+        if (error) console.log(error);
+    }
+})();
```
Comment (oesukam) [inline server/v1/models/db/schema.js:54]: Empty line required

File: server/v2/config/db.config.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+const connectionString = 'postgres://winner:winners1@localhost:5432/way_farer_db';
+const pool = new Pool({ 
+	user: 'winner',
+  	host: 'localhost',
+  	database: 'way_farer_db',
+  	password: 'winners1',
+  	port: 5432
+});
+
+export default pool;
```
Comment (oesukam) [inline server/v2/config/db.config.js]: Empty line required

File: server/v2/controllers/auth_controller.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        const token = isAdmin 
+        ? userHelper.authenticateUser({id,email,first_name, isAdmin})
+        : userHelper.authenticateUser({id,email,first_name});
+        
+        userHelper.respond(res, 200, "success", "signed in successfully", {token});
+    }
+
+    
+}
+export default authController
```
Comment (oesukam) [inline server/v2/controllers/auth_controller.js]: Empty line required

File: server/v2/controllers/bookings_controller.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    deleteBooking : (req, res) => {
+        Bookings.delete(req.params.booking_id);
+        userHelper.respond(res, 200, "success", "booking deleted successfully", { message : "Booking deleted successfully"});
+    },
+    viewBookings : (req, res) => {
+        const bookings = (req.user.is_admin) ? Bookings.findAll('bookings') : Bookings.findUserBookings(req.user.id);
+        userHelper.respond(res, 200, "success", "OK", bookings);
+    }
+}
+export default bookingsController
```
Comment (oesukam) [inline server/v2/controllers/bookings_controller.js]: Empty line required

File: server/v2/controllers/trips_controller.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+    },
+    filterTrips : (req, res) => {
+        const { origin, destination } = req.query;
+        const target = origin ? "origin" : "destination";
+        const trips = Trips.filterBy(target, (origin ? origin : destination));
+        userHelper.respond(res, 200, "success","trips filtered successfully", trips);
+    }
+}
+export default tripsController
```
Comment (oesukam) [inline server/v2/controllers/trips_controller.js]: Empty line required

File: server/v2/helpers/user_helper.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+		status : statusCode,
+		message,
+		data : payload
+	})
+	: res.status(statusCode).send({
+		status : statusCode,
+		error : payload
+	})
+}
+export default { hashPassword, comparePasswords, authenticateUser, respond };
```
Comment (oesukam) [inline server/v2/helpers/user_helper.js]: Empty line required

File: server/v2/middlewares/authentication.js [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+    else{
+        return done(null, false);
+    }
+}));
+
+export default passport;
+
+
+
+// RestFul Authentication with Passport
```
Comment (oesukam) [inline server/v2/middlewares/authentication.js]: Empty line required and unnecessary comment

File: server/v2/routes/index.js [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+router.use('/', tripsRouter);
+router.use('/', bookingsRouter);
+
+router.get('/', (req, res) => {
+    // eslint-disable-next-line max-len
+    res.status(200).send({message : `Welcome to WayFarer api feel home access the documentation ${docsUrl}`})
+})
+
+
+export default router
```
Comment (oesukam) [inline server/v2/routes/index.js]: Empty line required

File: server/v2/routes/trips.js [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+import tripsController from '../controllers/trips_controller';
+const routes = express.Router()
+
+routes
+.post('/trips', authValidations.checkUserToken, authValidations.isAdmin, tripsValidations.validateTrip, tripsController.createTrip)
+.patch('/trips/:trip_id/cancel', authValidations.checkUserToken, authValidations.isAdmin, tripsValidations.isValidTrip, tripsController.cancelTrip)
+.get('/trips/:trip_id', tripsValidations.isValidTrip, tripsController.viewTrip)
+.get('/trips/',  tripsController.viewTrips)
+.get('/filter/trips', tripsValidations.validateFilter, tripsController.filterTrips)
+export default routes;
```
Comment (oesukam) [inline server/v2/routes/trips.js]: Empty line required

File: server/v2/tests/bookings.test.js [hunk 12]
```diff
@@ -0,0 +1,296 @@
...
+          expect(res.body).to.be.an('object')
+          expect(res.body).to.have.property('data')
+          expect(res.body.data).to.be.an('array')
+          expect(res.body.status).to.equal(200)
+          done();
+        })
+  });
+
+
+})
```
Comment (oesukam) [inline server/v2/tests/bookings.test.js]: Empty line required

File: server/v2/tests/generator.js [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+            const newTrip = User.create(trip, 'trips');
+        });
+    },
+    generateBookings : () => {
+        generator.bookings.forEach(booking => {
+            const newBooking = User.create(booking, 'bookings');
+        });
+    }
+}
+export default generator;
```
Comment (oesukam) [inline server/v2/tests/generator.js]: Empty line required

File: server/v2/tests/trips.test.js [hunk 14]
```diff
@@ -0,0 +1,298 @@
...
+          if (err) done(err);
+          expect(res).to.have.status(200)
+          expect(res.body).to.have.property('data')
+          expect(res.body.data).to.be.an('array')
+          expect(res.body.status).to.equal(200)
+          done();
+        })
+  });
+
+})
```
Comment (oesukam) [inline server/v2/tests/trips.test.js]: Empty line required

File: server/v2/tests/users.test.js [hunk 15]
```diff
@@ -0,0 +1,181 @@
...
+        .get('/api/v1111/wrong')
+        .set('Content-type', 'application/json')
+        .end((err, res) => {
+          if (err) done(err);
+          expect(res).to.have.status(404);
+          expect(res.body.error).to.equal(`The requested resource was not found on the server, Kindly read documentation here : ${docsUrl}`);
+          done();
+        })
+    });
+})
```
Comment (oesukam) [inline server/v2/tests/users.test.js]: Empty line required

**Review by oesukam — CHANGES_REQUESTED** [review]

**Comment by WinnersProx** [comment]
I'm getting that but the problem is that I can't figure out where you want
me to put an empty line.Can you please help me to do it in a most
expressive way?

**Review by oesukam — CHANGES_REQUESTED** [review]
Add a blank line at the end of each file

**Comment by WinnersProx** [comment]
Thanks so much, hold on I'm working on it!

File: server/v2/tests/users.test.js (hunk 15 above)
Comment (WinnersProx) [reply server/v2/tests/users.test.js]: @oesukam is it now fixed, can it be merged?

**Review by oesukam — APPROVED** [review]

**Comment by oesukam** [comment]
@WinnersProx I have approved your changes

**Comment by WinnersProx** [comment]
Noted, thanks!

**Pull request merged**